tanjun/abc.py

```diff
@@ -397,15 +611,141 @@ async def edit_last_response(
             typing.Union[hikari.SnowflakeishSequence[hikari.PartialRole], bool]
         ] = hikari.UNDEFINED,
     ) -> hikari.Message:
-        raise NotImplementedError
+        """Edit the last response for this context.
+
+        Parameters
+        ----------
+        content : hikari.UndefinedOr[typing.Any]
+            The content to edit the response with.
+
+            If provided, the message contents. If
+            `hikari.undefined.UNDEFINED`, then nothing will be sent
+            in the content. Any other value here will be cast to a
+            `str`.
+
+            If this is a `hikari.embeds.Embed` and no `embed` nor `embeds` kwarg
+            is provided, then this will instead update the embed. This allows
+            for simpler syntax when sending an embed alone.
+
+            Likewise, if this is a `hikari.files.Resource`, then the
+            content is instead treated as an attachment if no `attachment` and
+            no `attachments` kwargs are provided.
+
+        Other Parameters
+        ----------------
+        attachment : hikari.UndefinedOr[hikari.Resourceish]
+            A singular attachment to edit the response with.
+        attachments : hikari.UndefinedOr[collections.Sequence[hikari.Resourceish]]
+            A sequence of attachments to edit the response with.
+        embed : hikari.UndefinedOr[hikari.Embed]
+            An embed to replace the response with.
+        embeds : hikari.UndefinedOr[collections.Sequence[hikari.Embed]]
+            A sequence of embeds to replace the response with.
+        replace_attachments : bool
+            Whether to replace the attachments of the response or not. Default to `False`.
+        mentions_everyone : hikari.undefined.UndefinedOr[bool]
+            If provided, whether the message should parse @everyone/@here
+            mentions.
+        user_mentions : hikari.undefined.UndefinedOr[typing.Union[hikari.snowflakes.SnowflakeishSequence[hikari.users.PartialUser], bool]]
+            If provided, and `True`, all mentions will be parsed.
+            If provided, and `False`, no mentions will be parsed.
+            Alternatively this may be a collection of
+            `hikari.snowflakes.Snowflake`, or `hikari.users.PartialUser`
+            derivatives to enforce mentioning specific users.
+        role_mentions : hikari.undefined.UndefinedOr[typing.Union[hikari.snowflakes.SnowflakeishSequence[hikari.guilds.PartialRole], bool]]
+            If provided, and `True`, all mentions will be parsed.
+            If provided, and `False`, no mentions will be parsed.
+            Alternatively this may be a collection of
+            `hikari.snowflakes.Snowflake`, or
+            `hikari.guilds.PartialRole` derivatives to enforce mentioning
+            specific roles.
+
+        .. note::
```

Comment:
Notes aren't supported in Parameters segment under pdoc, in this case it'd be cleaner to turn this into a
```
Notes
------
```
segment where it is right now (but note that usually i'd say to move it over the parameters segments if it's just one small note)

Reply:
Forgot a about this part. Will take a look at it.